hack/aws-acceptance-test-cleanup/main.go

```diff
@@ -68,6 +69,94 @@ func realMain(ctx context.Context) error {
 	eksClient := eks.New(clientSession, awsCfg)
 	ec2Client := ec2.New(clientSession, awsCfg)
 	elbClient := elb.New(clientSession, awsCfg)
+	iamClient := iam.New(clientSession, awsCfg)
+
+	// Find OIDC providers to delete.
+	oidcProvidersOutput, err := iamClient.ListOpenIDConnectProvidersWithContext(ctx, &iam.ListOpenIDConnectProvidersInput{})
+	type oidc struct {
+		arn      string
+		buildUrl string
+	}
+	toDeleteOidcArns := []*oidc{}
+	for _, providerEntry := range oidcProvidersOutput.OpenIDConnectProviderList {
+		arnString := ""
+		if providerEntry.Arn != nil {
+			arnString = *providerEntry.Arn
+		}
+		output, err := iamClient.ListOpenIDConnectProviderTags(&iam.ListOpenIDConnectProviderTagsInput{OpenIDConnectProviderArn: providerEntry.Arn})
+		if err != nil {
+			return err
+		}
+		for _, tag := range output.Tags {
+			if tag.Key != nil && *tag.Key == buildURLTag {
+				var buildUrl string
+				if tag.Value != nil {
+					buildUrl = *tag.Value
+				}
+				toDeleteOidcArns = append(toDeleteOidcArns, &oidc{arn: arnString, buildUrl: buildUrl})
+			}
+		}
+	}
+
+	oidcProvidersExist := true
+	if len(toDeleteOidcArns) == 0 {
+		fmt.Println("Found no OIDC Providers to clean up")
+		oidcProvidersExist = false
+	} else {
+		// Print out the OIDC Provider arns and the build tags.
+		var oidcPrint string
+		for _, oidcProvider := range toDeleteOidcArns {
+			oidcPrint += fmt.Sprintf("- %s (%s)\n", oidcProvider.arn, oidcProvider.buildUrl)
+		}
+
+		fmt.Printf("Found OIDC Providers:\n%s", oidcPrint)
+	}
+
+	// Check for approval.
+	if !flagAutoApprove && oidcProvidersExist {
+		type input struct {
+			text string
+			err  error
+		}
+		inputCh := make(chan input)
+
+		// Read input in a goroutine so we can also exit if we get a Ctrl-C
+		// (see select{} below).
+		go func() {
+			reader := bufio.NewReader(os.Stdin)
+			fmt.Println("\nDo you want to delete these OIDC Providers (y/n)?")
```

Comment:
It looks like this will be a manually run job instead of an automated one? Also, should we consider putting in a check and only deleting things older than 24 hours so we don't break tests that are currently in-flight?

Reply:
I'm not 100% on the history of why we have this "manual" looking step, but it's also in the other cleanup functions as well, so looks like @ndhanushkodi just used the same pattern that's established in the repo.

Reply:
The approval step is for running locally, so you can see all the things you're about to delete (and its tags) and approve the deletion. I added a separate approval step for this because IAM resources are separate from VPCs. The approval step is only prompted if you don't set --auto-approve, which our pipelines do when they run the script. 

Yeah the automated pipeline step should be safe to not interfere with the running job, but I also went ahead a timestamp check here anyways so if you run it manually you wouldn't delete the most recent oidc.

Reply:
Awesome, thank you.